bundles/org.openhab.binding.daikin/src/main/java/org/openhab/binding/daikin/internal/handler/DaikinBaseHandler.java

```diff
@@ -0,0 +1,202 @@
+import java.util.List;
+import java.util.ArrayList;
+
+import javax.measure.quantity.Temperature;
+
+import org.eclipse.smarthome.core.library.types.DecimalType;
+import org.eclipse.smarthome.core.library.types.OnOffType;
+import org.eclipse.smarthome.core.library.types.QuantityType;
+import org.eclipse.smarthome.core.library.types.StringType;
+import org.eclipse.smarthome.core.library.unit.SIUnits;
+import org.eclipse.smarthome.core.thing.ChannelUID;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingTypeUID;
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.thing.ThingStatusDetail;
+import org.eclipse.smarthome.core.thing.binding.BaseThingHandler;
+import org.eclipse.smarthome.core.types.Command;
+import org.eclipse.smarthome.core.types.State;
+import org.eclipse.smarthome.core.types.StateOption;
+import org.eclipse.smarthome.core.types.UnDefType;
+import org.openhab.binding.daikin.internal.DaikinBindingConstants;
+import org.openhab.binding.daikin.internal.DaikinCommunicationException;
+import org.openhab.binding.daikin.internal.DaikinWebTargets;
+import org.openhab.binding.daikin.internal.DaikinDynamicStateDescriptionProvider;
+
+import org.openhab.binding.daikin.internal.config.DaikinConfiguration;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Base class that handles common tasks with a Daikin air conditioning unit.
+ *
+ * @author Tim Waterhouse - Initial Contribution
+ * @author Paul Smedley <paul@smedley.id.au> - Modifications to support Airbase Controllers
+ * @author Jimmy Tanagra - Split handler classes, support Airside and DynamicStateDescription
+ *
+ */
+public abstract class DaikinBaseHandler extends BaseThingHandler {
+
+    protected final Logger logger;
+
+    private long refreshInterval;
+
+    protected DaikinWebTargets webTargets;
+    private ScheduledFuture<?> pollFuture;
+    protected final DaikinDynamicStateDescriptionProvider stateDescriptionProvider;
+
+    // Abstract methods to be overridden by specific Daikin implementation class
+    protected abstract void pollStatus() throws IOException;
+
+    protected abstract void changePower(boolean power) throws DaikinCommunicationException;
+    protected abstract void changeSetPoint(double newTemperature) throws DaikinCommunicationException;
+    protected abstract void changeMode(String mode) throws DaikinCommunicationException;
+    protected abstract void changeFanSpeed(String fanSpeed) throws DaikinCommunicationException;
+
+    // Power, Temp, Fan and Mode are handled in this base class. Override this to handle additional channels. 
+    protected abstract boolean handleCommandInternal(ChannelUID channelUID, Command command) throws DaikinCommunicationException;
+
+
+    public DaikinBaseHandler(Thing thing, DaikinDynamicStateDescriptionProvider stateDescriptionProvider) {
+        super(thing);
+        logger = LoggerFactory.getLogger(getClass());
```

Comment:
I think you probaly want to reuse the loggers in the sub-classes. Please don't do that and use seperate loggers. Include the logger instatntiation in the field defintion and make them private.

Reply:
You're correct, that was my intention. I will set this to private now and instantiate it in the field definition.